src/services/logger/DisabledLogger.ts

```diff
@@ -1,8 +1,17 @@
-import Logger from './DatadogLogger'
 import type { ClientConfigInterface } from '@/store/modules/config/config.types'
-
-export default class DisabledLogger extends Logger {
-  setup(_config: ClientConfigInterface) {
-    console.log('Logging disabled')
+const c = class {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(..._args: any[]) {
   }
 }
+export const disabledLogger = function (parent: typeof c) {
+  const name = `Disabled${parent.name}`
+  const disabled = {
+    [name]: class extends parent {
+      setup(_config: ClientConfigInterface) {
+        console.log('Logging disabled')
+      }
+    },
+  }
+  return disabled[name]
```

Comment:
**Suggestion**: I’m not sure why you’re first creating an object with a named key only to return the value assigned to that key. Wouldn’t the following be equivalent?

```suggestion
  return class extends parent {
    setup(_config: ClientConfigInterface) {
      console.log('Logging disabled')
    }
  }
```

Same suggestion applies to `MockKumaApi`.

Reply:
It makes sure we keep a name during debugging. Honestly I don't really mind either way, in fact I'd be happier with the suggestion you made here, but in other PRs you made a clear preference to try to ensure we use named functions for debugging purposes, and this ensures its named. Happy to remove it if you don't want it, it was only for your benefit.

Reply:
> in other PRs you made a clear preference to try to ensure we use named functions for debugging purposes,

I did that in the very specific context of a warning that made use of a function’s name so this doesn’t apply here.

I’d rather have the simpler version.

Reply:
just to check you are fine me using this form right?

```javascript
return class DisabledLogger extends parent ()
///
return class MockApi extends parent ()
```

i.e. statically named classes instead of dynamic? Let me know either way, I'm good with or without.

Reply:
Yeah that’s fine

Reply:
Ok just a quick ping to say this is done, will merge in 5mins or so if I hear nothing back 👍